cloud/deploy/deploy.go

```diff
@@ -59,84 +65,142 @@ type deploymentInfo struct {
 	webserverURL   string
 }
 
-// Deploy pushes a new docker image
-func Deploy(path, deploymentID, wsID, pytest, envFile, imageName string, prompt bool, client astro.Client) error {
-	// Get cloud domain
-	c, err := config.GetCurrentContext()
+func deployDags(path, domain string, deployInfo *deploymentInfo, client astro.Client) error {
+	fmt.Println("Initiating DAGs Deployment for: " + deployInfo.deploymentID)
+	dagDeployment, err := deployment.Initiate(deployInfo.deploymentID, client)
 	if err != nil {
 		return err
 	}
 
-	cloudDomain := c.Domain
-	if cloudDomain == "" {
-		return errors.New("no domain set, re-authenticate")
-	}
+	// Check the dags directory
+	dagsPath := path + "/dags"
 
-	deployInfo, err := getDeploymentInfo(deploymentID, wsID, prompt, cloudDomain, client)
+	// Generate the dags tar
+	err = fileutil.Tar(dagsPath, "/tmp")
 	if err != nil {
 		return err
 	}
 
-	// Build our image
-	version, err := buildImage(&c, path, deployInfo.currentVersion, deployInfo.deployImage, imageName, client)
+	sasDagClient, err := azure.CreateSASDagClient(dagDeployment.DagURL)
 	if err != nil {
-		return err
+		log.Fatal(err)
 	}
 
-	err = parseDAG(pytest, version, envFile, deployInfo.deployImage, deployInfo.namespace)
+	dagFile, err := os.Open("/tmp/dags.tar")
 	if err != nil {
 		return err
 	}
-
-	// Create the image
-	imageCreateInput := astro.ImageCreateInput{
-		Tag:          version,
-		DeploymentID: deployInfo.deploymentID,
-	}
-	imageCreateRes, err := client.CreateImage(imageCreateInput)
+	versionID, err := sasDagClient.Upload(dagFile)
 	if err != nil {
 		return err
 	}
 
-	domain := c.Domain
-	if strings.Contains(domain, "cloud") {
-		splitDomain := strings.SplitN(domain, ".", splitNum) // This splits out 'cloud' from the domain string
-		domain = splitDomain[1]
+	var status string
+	if versionID != "" {
+		status = "SUCCEEDED"
+	} else {
+		status = "FAILED"
 	}
 
-	nextTag := "deploy-" + time.Now().UTC().Format("2006-01-02T15-04")
-	var registry string
-	if domain == "localhost" {
-		registry = config.CFG.LocalRegistry.GetString()
-	} else {
-		registry = "images." + strings.Split(domain, ".")[0] + ".cloud"
+	_, err = deployment.ReportDagDeploymentStatus(dagDeployment.ID, deployInfo.deploymentID, action, versionID, status, message, client)
+	if err != nil {
+		return err
 	}
-	repository := registry + "/" + deployInfo.organizationID + "/" + deployInfo.deploymentID
-	// TODO: Resolve the edge case where two people push the same nextTag at the same time
-	remoteImage := fmt.Sprintf("%s:%s", repository, nextTag)
 
-	token := c.Token
-	// Splitting out the Bearer part from the token
-	splittedToken := strings.Split(token, " ")[1]
+	deploymentURL := "cloud." + domain + "/" + deployInfo.organizationID + "/deployments/" + deployInfo.deploymentID
```

Comment:
nitpick: not directly related to this PR, but can we append `deploymentURL` with `https://` it will allow cmd+click to redirect and open the URL in browser instead of the current way of copy-pasting the URL from terminal to browser

<img width="997" alt="Screenshot 2022-07-25 at 6 28 43 PM" src="https://user-images.githubusercontent.com/92356010/180782965-39f7e677-e6a2-49d9-9e1f-a2c17c7e44c3.png">

Reply:
Good call. Making a note of this. Will add this as a separate PR

Reply:
@neel-astro Seems like we can still do `cmd+click` on the above urls. We dont really need the `https://`